pkg/acquisition/journalctl_reader.go

```diff
@@ -0,0 +1,174 @@
+package acquisition
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os/exec"
+	"strings"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	leaky "github.com/crowdsecurity/crowdsec/pkg/leakybucket"
+	"github.com/crowdsecurity/crowdsec/pkg/types"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	tomb "gopkg.in/tomb.v2"
+)
+
+/*
+ journald/systemd support :
+
+ systemd has its own logging system, which stores files in non-text mode.
+ To be able to read those, we're going to read the output of journalctl, see https://github.com/crowdsecurity/crowdsec/issues/423
+
+
+ TBD :
+  - handle journalctl errors
+*/
+
+type JournaldSource struct {
+	Config  DataSourceCfg
+	Cmd     *exec.Cmd
+	Stdout  io.ReadCloser
+	Stderr  io.ReadCloser
+	Decoder *json.Decoder
+	SrcName string
+}
+
+var JOURNALD_CMD = "journalctl"
+var JOURNALD_DEFAULT_TAIL_ARGS = []string{"--follow"}
+var JOURNALD_DEFAULT_CAT_ARGS = []string{}
+
```

Comment:
IMHO these shouldn't be variables, but constants, to ensure they are immutable.

Reply:
yes they should stay as variables: unit tests use those to mock journalctl execution